x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -230,31 +228,18 @@ export const useCreateAnalyticsForm = () => {
 
     try {
       // Set the index pattern titles which the user can choose as the source.
-      const indexPatternTitles = await kibanaContext.indexPatterns.getTitles(true);
-      // Find out which index patterns contain numeric fields.
-      // This will be used to provide a hint in the form that an analytics jobs is not
-      // able to identify outliers if there are no numeric fields present.
-      const ids = await kibanaContext.indexPatterns.getIds(true);
-      const indexPatternsWithNumericFields: IndexPatternTitle[] = [];
-      const indexPatternsMap = {}; // TODO: add type, add to state to keep track of
-      ids
-        .filter(f => !!f)
-        .forEach(async id => {
-          const indexPattern = await kibanaContext.indexPatterns.get(id!);
-          if (
-            indexPattern.fields
-              .filter(f => !OMIT_FIELDS.includes(f.name))
-              .map(f => f.type)
-              .includes('number')
-          ) {
-            indexPatternsWithNumericFields.push(indexPattern.title);
-            // @ts-ignore TODO: fix this type
-            indexPatternsMap[indexPattern.title] = id;
-          }
-        });
+      const indexPatternTitles: string[] = [];
+      const indexPatternsMap: { [key: string]: string | undefined } = {};
```

Comment:
You can define it as `Record<string, string | undefined>` (and in `Action` type as well )

Reply:
Updated here b2e501c17ea72ad79e742cba55bb69bf75417f44